### PR TITLE
feat(appwrite): add support to conditional filters and missing logical filters

### LINKED ISSUE #6252: [FEAT] Add support to `and`, `or`, `between`, `null` and `nnull` operators in `@refinedev/appwrite`
### Is your feature request related to a problem? Please describe.

`and` and `or` are common operators in filtering.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Describe the thing to improve

Please assign it to me ❤️

### PR DESCRIPTION
**Pull request by soranoo**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://refine.dev/docs/guides-concepts/contributing/#commit-convention

## Features

- [x] Related issue(s) linked https://github.com/refinedev/refine/issues/6252
- [x] Tests for the changes have been added
- [x] Docs have been added / updated
- [x] Changesets have been added https://refine.dev/docs/guides-concepts/contributing/#creating-a-changeset

## What is the current behavior?
Missing `and`, `or`, `between`, `null`, `nnull`, `startswith` and `endswith` operators

## What is the new behavior?
Able to use `and`, `or`, `between`, `null`, `nnull`, `startswith` and `endswith` operators

### CODE REVIEW

File: packages/appwrite/src/utils/getAppwriteFilters.ts [hunk 1]
```diff
@@ -1,26 +1,14 @@
...
-      const filterField = filter.field === "id" ? "$id" : filter.field;
-
-      appwriteFilters.push(
-        generateFilter({
-          ...filter,
-          field: filterField,
-        }),
-      );
-    }
+    appwriteFilters.push(generateFilter(replaceIdWithAppwriteId(filter)));
```
Comment (aliemir) [inline packages/appwrite/src/utils/getAppwriteFilters.ts:11]: It's great to see this part is finally refactored to a better shape 😅 Do you think we can move `replaceIdWithAppwriteId` call to `generateFilter` method? I'm not sure if Appwrite supports nested conditional filters (`or`, `and`) but if they do we'll need to check for them too, am I wrong? 🤔 

We can check their docs and implementation or test it manually and see if they support it. We can also add a test case to see if we cover that.

What do you think @soranoo? 🙏

**Review by aliemir — COMMENTED** [review]
Hey @soranoo thank you for the PR, it looks great! 🚀 🚀  I've left a small comment, I hope you can check it 🙏

File: packages/appwrite/src/utils/getAppwriteFilters.ts (hunk 1 above)
Comment (soranoo) [reply packages/appwrite/src/utils/getAppwriteFilters.ts:11]: Yes, Appwrite is support nest `or` and `and`
image
ref: https://appwrite.io/docs/products/databases/queries#query-class

Comment (soranoo) [reply packages/appwrite/src/utils/getAppwriteFilters.ts:11]: yes, Appwrite does support nested conditional filters, just checked it.

**Review by aliemir — APPROVED** [review]

**Pull request merged**